src/macros.asm

```diff
@@ -27,6 +27,24 @@ MACRO enemy x, y, type
 	.db type, x << 4 | y
 ENDM
 
+;
+; LevelHeader macro
+;
+; The order of the parameters is slightly different than how it's encoded, but
+; hopefully this order is a little more intuitive?
+;
+MACRO levelHeader pages, horizontal, bgPalette, spritePalette, music, objectTypeAXFX, objectType3X9X, groundSetting, groundType
+	.db horizontal << 7 | bgPalette << 3 | spritePalette
+	.db %11100000 | groundSetting
+	.db pages << 4 | objectTypeAXFX << 2 | objectType3X9X
+	IFNDEF LEVEL_ENGINE_UPGRADES
+		.db groundType << 3 | music
+	ENDIF
+	IFDEF LEVEL_ENGINE_UPGRADES
+		.db groundType << 4 | music
```

Comment:
`xxTTTxMM` in vanilla, vs `TTTTxxMM` with the flag, which allows twice as many ground types plus two contiguous bytes for something else (eg. `LevelObjectType`)

interestingly, it seems like the F6 objects are already set up to use 4-bit for ground types!

Reply:
Probably worth noting that potential oddity in the source -- do we have a standard comment like that yet? I'm thinking about the Mario 3 disasm documentation and its use of "points of interest", so maybe something like `; POI: Seems like this was built to expect 4-bit types?` ... Would have to go back and add this in other places, though. Could make this an official thing in a contributing/style guide, hah